classes/ObjectModel.php

```diff
@@ -1134,18 +1134,21 @@ public function validateController($htmlentities = true)
             // Checking for fields validity
             // Hack for postcode required for country which does not have postcodes
             if (!empty($value) || $value === '0' || ($field == 'postcode' && $value == '0')) {
-                if (isset($data['validate']) && !Validate::$data['validate']($value) && (!empty($value) || $data['required'])) {
-                    $errors[$field] = '<b>'.self::displayFieldName($field, get_class($this), $htmlentities).'</b> '.Tools::displayError('is invalid.');
-                } else {
-                    if (isset($data['copy_post']) && !$data['copy_post']) {
-                        continue;
-                    }
-                    if ($field == 'passwd') {
-                        if ($value = Tools::getValue($field)) {
-                            $this->{$field} = Tools::encrypt($value);
-                        }
+                if (isset($data['validate'])) {
+                    $data_validate = $data['validate'];
+                    if (!Validate::$data_validate($value) && (!empty($value) || $data['required'])) {
+                        $errors[$field] = '<b>'.self::displayFieldName($field, get_class($this), $htmlentities).'</b> '.Tools::displayError('is invalid.');
```

Comment:
I checked the diff a new time and something is wrong here.

It seems the test structure has changed. Instead of 
```php
if (isset($data['validate'])) {
    [...]
    if (!Validate::$data_validate($value) && (!empty($value) || $data['required'])) {
        [...]
    } else {
        if (isset($data['copy_post']) && !$data['copy_post']) {
            continue;
        }
    }
}
```

we should have
```php
if (isset($data['validate'])) {
    [...]
    if (!Validate::$data_validate($value) && (!empty($value) || $data['required'])) {
        [...]
    }
} else {
    if (isset($data['copy_post']) && !$data['copy_post']) {
        continue;
    }
}
```

Do you see what I mean ?
When this point will be fixed, @julienbourdeau will be able to merge this PR.

Best regards